src/org/opendatakit/briefcase/model/ServerConnectionInfo.java

```diff
@@ -69,11 +70,11 @@ public HttpClient getHttpClient() {
   }
 
   public HttpClientContext getHttpContext() {
```

Comment:
What do you think about naming these get/setThreadHttpClientContext, to add clarity. And doc comment(s) to explain what’s happening and why it’s necessary. 

In one of the callers, there’s a comment which this makes less correct:
```
    // get shared HttpContext so that authentication and cookies are retained.
    HttpClientContext localContext = serverInfo.getHttpContext();
```

Reply:
Thanks for the feedback @dcbriccetti. 

To me, those comments are just as valid after making the the context local to the thread. You want multiple requests to share take advantage of previous contextual information, otherwise you would make it local to the method instead. The key is making the scope clear, which I feel the original code conflates since both context and client creation are often done together (which isn't necessary, nor idiomatic usage of http components). 

As for the naming suggestion, it does make the scope difference more obvious to the client code. However, in the class, it's already obvious by the member name and type.  I am not convinced the method name change is necessary. There is precedent for not baking "Thread" into the method name for thread-bound object lookups. Hibernate's getCurrentSession(), JPA EntityManager lookup, or Spring's and JavaEE's transaction lookup calls, are a few examples.

However, a bit of quality javadoc documentation or, even better, reorganization so objects better match their real responsibilities will make the code easier to understand.

As a side note, in a local branch, I eliminated the setter for the HttpClientContext class. That method enables assignment of so multiple threads can still share the same reference. With the change, the context can only be shared if explicitly doing so in client code.

Reply:
Merged since @dcbriccetti's comment isn't directly related to the fix. But better comments and/or separation of responsibilities would of course be fantastic. @dcbriccetti maybe when you have a chance you could add the comments you had in mind?

Reply:
@lognaturel, I have some additional changes that should simplify the intention a bit. It might just be a little bit since they are embedded in local work I am doing for concurrent push. @dcbriccetti don't let that stop you from improving it. I am happy to go with the flow.